pstatsplugin.cpp

```diff
@@ -74,23 +82,50 @@ UR_FIELDS (
 PSTATSPlugin::PSTATSPlugin(const options_t &module_options)
 {
    print_stats = module_options.print_stats;
+   use_zeros = false;
+}
+
+void PSTATSPlugin::check_plugin_options(vector<plugin_opt>& plugin_options)
+{
+   stringstream rawoptions(plugin_options[0].params);
+   string option;
+   vector<string> options;
+
+   while (std::getline(rawoptions, option, ':')) {
+      std::transform(option.begin(), option.end(), option.begin(), ::tolower);
+      options.push_back(option);
+   }
+
+   for (size_t i = 0; i < options.size(); i++) {
+      if (options[i] == INCLUDE_ZEROS_OPT) {
+         DEBUG_MSG("PSTATS include zero-length packets\n");
+         use_zeros = true;
+      }
+   }
 }
 
 PSTATSPlugin::PSTATSPlugin(const options_t &module_options, vector<plugin_opt> plugin_options) : FlowCachePlugin(plugin_options)
 {
    print_stats = module_options.print_stats;
+   use_zeros = false;
+   check_plugin_options(plugin_options);
+
 }
 
 void PSTATSPlugin::update_record(RecordExtPSTATS *pstats_data, const Packet &pkt)
 {
+  if(pkt.payload_length == 0 && use_zeros == false){
+    return;
+  }
+
    /*
     * dir =  1 iff client -> server
     * dir = -1 iff server -> client
     */
    int8_t dir = pkt.source_pkt ? 1 : -1;
    if (pstats_data->pkt_count < PSTATS_MAXELEMCOUNT) {
       uint16_t pkt_cnt = pstats_data->pkt_count;
-      pstats_data->pkt_sizes[pkt_cnt] = pkt.ip_length;
+      pstats_data->pkt_sizes[pkt_cnt] = pkt.payload_length;
```

Comment:
Shouldn't this be a `payload_length_orig` (payload length computed from headers) instead of, most likely trimmed, captured payload length?

Reply:
Yes, you are probably right. I did not consider packet trimming.